Fix Lint/FormatParameterMismatch for splatted last argument

zverok opened the pull request:
Previously, cases like `'#<%s (%s-%s)>' % [self.class, *values.minmax]` gave false offense (too low argument number) for this cop. With this commit, only when provided arguments count is definitely more than expected (even considering splat), offense is generated:

```ruby
# code
array = [1, 2]
puts '#<%s (%s-%s)>' % [1, *array]
# rubocop: no offenses

# code
array = [1, 2]
puts '#<%s>' % [1, 2, *array]
# rubocop: 
# Number of arguments (-2) to String#% doesn't match the number of fields (1).
```
Negative arguments count report, to be honest, is not the most intuitive, yet in line with Ruby's `Method#arity`, which is negative with splat.

bbatsov commented on CHANGELOG.md:
```diff
@@ -46,6 +46,7 @@
 * [#3815](https://github.com/bbatsov/rubocop/pull/3815): Fix false positive in `Style/IdenticalConditionalBranches` cop when branches have same line at leading. ([@pocke][])
 * Fix false negative in `Rails/HttpPositionalArguments` where offense would go undetected if one of the request parameter names matched one of the special keyword arguments. ([@deivid-rodriguez][])
 * Fix false negative in `Rails/HttpPositionalArguments` where offense would go undetected if the `:format` keyword was used with other non-special keywords. ([@deivid-rodriguez][])
+* Fix `Lint/FormatParameterMismatch` for splatted last argument. ([@zverok][])
```
Did you add yourself to the bottom of the file?

bbatsov commented on spec/rubocop/cop/lint/format_parameter_mismatch_spec.rb:
```diff
@@ -138,6 +138,24 @@
     expect(cop.offenses).to be_empty
   end
 
+  it 'does not register an offense on argument splat' do
```
This description could be better. Now it reads like it never registers offences if a splat arg is present.

bbatsov commented:
Apart from my small remark the changes look good. You'll also need to rebase.

zverok commented:
@bbatsov I've fixed the description (by providing common context) and rebased, but not sure I am getting your comment about changelog. I've added my change description to the end of `master (unreleased)`'s section "Bug fixes" (though, maybe it should be listed as "Changes" instead?)

bbatsov commented:
@zverok Scroll to the end of the changelog and you'll understand immediately why the build fails. :-)

zverok commented:
Well, it was dumb :( Fixed, I guess.

bbatsov commented:
👍
